src/Voron/Data/Graphs/Hnsw.cs

```diff
@@ -417,6 +423,47 @@ private void LoadNodeIndexes(ref NativeList<long> nodeIds, ref NativeList<int> i
                 reader.LoadInto(ref _nodes[indexes[matches + i]]);
             }
         }
+        
+        private void LoadNodeIndexes(List<long> nodeIds, List<int> indexes)
+        {
+            indexes.Clear();
+            indexes.EnsureCapacity(nodeIds.Count);
+            for (int i = 0; i < nodeIds.Count; i++)
+            {
+                if (_nodeIdToIdx.TryGetValue(nodeIds[i], out var index))
```

Comment:
We can use CollectionMarshall to retrieve the index

Reply:
No longer relevant